src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -63,120 +64,218 @@ public FetchAndMergeEntry(BibDatabaseContext bibDatabaseContext,
+        } else if (exception instanceof FetcherServerException) {
+            dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("Server not available"));
+        } else {
+            dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("Error occurred %0", exception.getMessage()));
+        }
+    }
+
+    private void mergeWithoutDialog(BibEntry originalEntry, BibEntry fetchedEntry) {
+        NamedCompound ce = new NamedCompound(Localization.lang("Merge entry without user interaction"));
+
+        updateEntryTypeIfDifferent(originalEntry, fetchedEntry, ce);
+        updateFieldsWithNewInfo(originalEntry, fetchedEntry, ce);
+        removeObsoleteFields(originalEntry, fetchedEntry, ce);
+
+        finalizeMerge(ce, originalEntry);
+    }
+
+    private void updateFieldsWithNewInfo(BibEntry originalEntry, BibEntry fetchedEntry, NamedCompound ce) {
+        Set<Field> jointFields = getJointFields(originalEntry, fetchedEntry);
+        for (Field field : jointFields) {
+            updateFieldIfNecessary(originalEntry, fetchedEntry, field, ce);
+        }
+    }
+
+    private void updateFieldIfNecessary(BibEntry originalEntry, BibEntry fetchedEntry, Field field, NamedCompound ce) {
+        fetchedEntry.getField(field).ifPresent(fetchedValue -> {
+            Optional<String> originalValue = originalEntry.getField(field);
+            if (originalValue.isEmpty() || fetchedValue.length() > originalValue.get().length()) {
+                originalEntry.setField(field, fetchedValue);
+                ce.addEdit(new UndoableFieldChange(originalEntry, field, originalValue.orElse(null), fetchedValue));
             }
+        });
+    }
+
+    private void removeObsoleteFields(BibEntry originalEntry, BibEntry fetchedEntry, NamedCompound ce) {
+        Set<Field> jointFields = getJointFields(originalEntry, fetchedEntry);
+        Set<Field> originalFields = getFields(originalEntry);
+
+        for (Field field : originalFields) {
+            if (!jointFields.contains(field) && !FieldFactory.isInternalField(field)) {
+                removeField(originalEntry, field, ce);
+            }
+        }
+    }
+
+    private void removeField(BibEntry entry, Field field, NamedCompound ce) {
+        Optional<String> originalValue = entry.getField(field);
+        entry.clearField(field);
+        ce.addEdit(new UndoableFieldChange(entry, field, originalValue.orElse(null), null));
+    }
+
+    private void finalizeMerge(NamedCompound ce, BibEntry entry) {
+        String citationKey = entry.getCitationKey().orElse(entry.getAuthorTitleYear(40));
+        String message = ce.hasEdits()
+                ? Localization.lang("Updated entry with fetched information [%0]", citationKey)
+                : Localization.lang("No new information was added [%0]", citationKey);
```

Comment:
Is this output for each entry?

Imagine a user updates 100 entries - are there 100 messages? How should the user understand this data?

Maybe better to have a single notification with: entries X udpates, where X is a list of the citation keys.

Even better it would be to have a magic JabRef group which is called "JR - updated", where all modified entries are added to. Before the action, all entries are removed from this group. (Not sure if you have time for that)